examples/cmd/snapshotting/example_demo.go

```diff
@@ -0,0 +1,410 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package main
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+	"time"
+
+	"golang.org/x/crypto/ssh"
+	"golang.org/x/sys/unix"
+
+	sdk "github.com/firecracker-microvm/firecracker-go-sdk"
+	models "github.com/firecracker-microvm/firecracker-go-sdk/client/models"
+)
+
+// Using default cache directory to ensure collision avoidance on IP allocations
+const cniCacheDir = "/var/lib/cni"
+const networkName = "fcnet"
+const ifName = "veth0"
+
+const networkMask string = "/24"
+const subnet string = "10.168.0.0" + networkMask
+
+const maxRetries int = 10
+const backoffTimeMs time.Duration = 500
+
+var ipToRestore string // Don't use global variables in production!
```

Comment:
I'm thinking rather than keeping this as a global var you could have `createSnapshotSSH` return the ip address to restore. Then, in main, you can get this ip from calling `createSnapshotSSH` and pass that into your load snapshot function